Set correct shared library name.

Don't use -A -n flags to strip (broken by default due to ports infrastructure).

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Index: configure
--- configure.orig
+++ configure
@@ -10999,7 +10999,7 @@ esac
     case "$target_cpu" in #(
   aarch64) :
 
-	for opt in -mbranch-protection=pac-ret -msign-return-address=all
+	for opt in -mbranch-protection=standard -msign-return-address=all
 do :
 
 
@@ -31462,7 +31462,7 @@ main (void)
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
-  if "${STRIP}" -A -n conftest$ac_exeext 2>/dev/null
+  if false # "${STRIP}" -A -n conftest$ac_exeext 2>/dev/null
 then :
 
             { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: -A -n" >&5
@@ -32010,7 +32010,7 @@ fi
 	 ;; #(
   openbsd*|mirbsd*) :
 
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby33_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -34136,7 +34136,7 @@ then :
 
 else $as_nop
 
-        arch="${target_cpu}-${target_os}"
+        arch="${target_cpu}-openbsd"
 
 fi
     printf "%s\n" "#define RUBY_PLATFORM \"$arch\"" >>confdefs.h
